expression/builtin_arithmetic.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"math"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/parser/opcode"
+	"github.com/pingcap/tidb/util/types"
+)
+
+var (
+	_ functionClass = &arithmeticPlusFunctionClass{}
+	_ functionClass = &arithmeticFunctionClass{}
+)
+
+var (
+	_ builtinFunc = &builtinArithmeticPlusRealSig{}
+	_ builtinFunc = &builtinArithmeticPlusIntSig{}
+	_ builtinFunc = &builtinArithmeticPlusDecimalSig{}
+	_ builtinFunc = &builtinArithmeticSig{}
+)
+
+type arithmeticPlusFunctionClass struct {
+	baseFunctionClass
+}
+
+// getEvalTp will get the `evalTp` of the argument according to it's return type.
+func (c *arithmeticPlusFunctionClass) getEvalTp(ft *types.FieldType) evalTp {
+	switch ft.Tp {
+	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong,
```

Comment:
no, for TypeDuration, `types.ClassType` will return `ClassString`, but we need `ClassInt` here.